.github/workflows/tests.yml

```diff
@@ -494,6 +494,240 @@ jobs:
           name: lxd-clients-${{ runner.os }}
           path: bin/
 
+  ui:
+    name: UI e2e tests
+    runs-on: ubuntu-latest
+    needs: [code-tests, documentation]
+    if: "!startsWith(github.ref, 'refs/heads/stable-')"
+    env:
+      LXD_DIR: "/var/lib/lxd"
+      LXD_OIDC_CLIENT_ID: "gxj297yfAjmklILK5WqPWDSbtVBAeSQm"
+      LXD_OIDC_ISSUER: "https://dev-xjrvvfikbsv4jxn7.us.auth0.com/"
+      LXD_OIDC_AUDIENCE: "https://dev-xjrvvfikbsv4jxn7.us.auth0.com/api/v2/"
+      LXD_OIDC_USER: "lxd-ui-e2e-tests@example.org"
+      LXD_OIDC_PASSWORD: "lxd-ui-e2e-password"
```

Comment:
This shouldnt be hardcoded into the test file, as it references an external service, and thus is a real password.

We should now change this password to something else and then securely transfer it to me so we can set it up as a GH secret.

Reply:
This is a dedicated free account, specially created for e2e tests. It is expected to be public. I don't see a reason for this to be kept secret.

Though once we start using the local OIDC service, we can remove this.

Reply:
Hrm, yeah, but we shouldnt be making the credentials public

Reply:
Created a new set of credentials and shared them to be setup as a repo secret. Once they are available, we can start using them here instead of the old ones.

Reply:
Thanks will set them up now.

As the creds will only be available after a PR is merged, please can you add a condition that this UI test only runs on push rather than pull requests. This way it wont cause our per-PR tests to take longer, but will still gate our edge builds.

Reply:
Adjusted the flow to rely on secrets.